CRM/Contribute/BAO/Contribution.php

```diff
@@ -539,6 +540,22 @@ public static function create(&$params) {
         $activityParams['source_contact_id'] = (int) ($params['source_contact_id'] ?? (CRM_Core_Session::getLoggedInContactID() ?: $contribution->contact_id));
         $activityParams['target_contact_id'] = ($activityParams['source_contact_id'] === (int) $contribution->contact_id) ? [] : [$contribution->contact_id];
       }
+      else {
+        // Check if target contact exists
+        $targetContactQuery = ActivityContact::get(FALSE)->setLimit(1)->setWhere([
+          ['activity_id', '=', $activityParams['id']],
+          ['record_type_id:name', '=', 'Activity Targets'],
+        ])->execute();
+
+        if ($targetContactQuery->rowCount === 0) {
+          // If no target contact exists, update source contact
+          $activityParams['source_contact_id'] = $contribution->contact_id;
+        }
+        else {
+          // If a target contact exists, update it
+          $activityParams['target_contact_id'] = $contribution->contact_id;
```

Comment:
On line 541 above, `$activityParams['target_contact_id']` accepts an array; here, it's passed a scalar.  Are both correct, or should this be an array?

Reply:
Thanks for your feedback!

Yes, I think this should be an array.